### ISSUE TITLE
New Terminal output adding spaces and % resulting in cut off when activating Python environments (venv)

### ISSUE THREAD

**Issue by slimsnerdy** [microsoft/vscode#192018]
Does this issue occur when all extensions are disabled?: Yes/No
Yes (only Python/Pylance installed since needed)

- VS Code Version: 1.81.1
- OS Version: macOS Ventura 13.5.1

Steps to Reproduce:

1.  Open Command Palette --> Python: Create Environment --> Venv --> Python (I chose Python 3.10.12 64-bit) --> requirements.txt (leave unchecked) click OK

2. Now create a new terminal; this is the strange part.. Sometimes it will add a % and a whole bunch of spaces when "activating" the environment.. look at the line with `source`.. `the_slim_fatty_p%                                               rod/.venv/bin/activate`
That is actually a bunch of spaces. Right click the Terminal window and "Toggle Size to Content Width" to see.

Sometimes creating a new terminal does not do this behavior and no spaces are added and that weird %.
This happens on new machines too. New projects from scratch. Python and macOS only tested.
Very random behavior; quickest way to test is click the plus icon on the bottom right to open new terminals so it activates in rapid succession and most of the time it actually does not add the spaces to the output. But every 4 or 5 clicks it does.. I have no idea.

![Screenshot 2023-08-24 at 11 14 56](https://github.com/microsoft/vscode/assets/5614558/64eeb75b-acb0-48c7-bd31-dd786c4eb346)

CORRECT BEHAVIOR:
![Screenshot 2023-08-24 at 11 00 02](https://github.com/microsoft/vscode/assets/5614558/2d5d073a-4586-4a07-9d67-8e1d968142db)

INCORRECT BEHAVIOR:
![Screenshot 2023-08-24 at 10 59 43](https://github.com/microsoft/vscode/assets/5614558/6a30e361-097d-4ca9-8b35-12688ab1f1d8)

```[tasklist]
### Tasks
```

**Comment by slimsnerdy** [comment]
> > warning] Microvenv failed as it is using proposed API which is constantly changing [TypeError: this.context.getEnvironmentVariableCollection is not a function
> 
> That issue should go away once you update your pre-release version of the extension., it's duplicate of [microsoft/vscode-python#21831](https://github.com/microsoft/vscode-python/issues/21831).

thank you; I will check

**Comment by Tyriar** [comment]
This is happening because you're using sendText before the shell is initialized. The only way to fix this is via a timeout (which is unreliable) or via a yet to come [executeCommand API](https://github.com/microsoft/vscode/issues/145234)

**Comment by karrtikr** [comment]
FWIW we do wait on terminal to open:

![image](https://github.com/microsoft/vscode/assets/13199757/770e0fd9-bd8c-4fbd-a5ce-d34bca7dc138)

Regardless, this issue should not be happening with https://github.com/microsoft/vscode-python/issues/11039, closing this.
